### ISSUE TITLE
[github-actions] Tagging GitHub Actions located in gatsby repo

### ISSUE THREAD

**Issue by lannonbr** [gatsbyjs/gatsby#16144]
## Summary

We've been using [GitHub Actions](https://github.com/features/actions) to help manage the repository and various aspects around it such as [maintaining the site showcase](https://github.com/gatsbyjs/gatsby/tree/master/.github/actions/gatsby-site-showcase-validator) and bringing up [high-priority PRs](https://github.com/gatsbyjs/gatsby/tree/master/.github/actions/high-priority-prs). Behind Actions, it is powered by Docker containers to spin up environments to run these tools. Currently for these Actions, we do not use a particular tag for these containers.

Those in the Docker community have commented that not tagging a container or using [latest can cause issues](https://vsupalov.com/docker-latest-tag/)

I would say with this, with how we tag and version packages / plugins for Gatsby, it would be beneficial of us to version the actions we have to help manage the repository. Then if we push some changes to one of the Actions, we can have a safety net before we "deploy" it to be used on the repo.

**Comment by mtheoryx** [comment]
Lannon is exactly right. I will only add that without proper docker image tags, it can be much more difficult to "rollback" from a bad deploy/build without a tagged, immutable container to choose from.

**Comment by lannonbr** [comment]
@pieh and I talked today and we decided given our actions are specific to us and tightly controlled, for now not tagging them should be alright. That said, if someone came up to us and wished to use something like the `high-priority-prs` action in their repo, then we could look into genralizing it and tagging it so it can be used outside our repo.

Alongside this, if we choose to use 3rd-party github actions in the future, this should be taken into consideration of using tagged verisions as we wouldn't be the maintainers of the actions.
